File: src/main/java/uk/co/lukestevens/logging/Logger.java
package uk.co.lukestevens.logging;

/**
 * An interface to define a simple logger
 * 
 * @author luke.stevens
 * @see LogManager
 */
public interface Logger {
	
	/**
	 * Logs a message under the INFO LoggerLevel
	 * @param message The message to log
	 */
	public void info(String message);
	
	/**
	 * Logs an exception under the INFO LoggerLevel
	 * @param e The exception to log
	 */
	public void info(Exception e);
	
	/**
	 * Logs a message under the DEBUG LoggerLevel
	 * @param message The message to log
	 */
	public void debug(String message);
	
	/**
	 * Logs an exception under the DEBUG LoggerLevel
	 * @param e The exception to log
	 */
	public void debug(Exception e);
	
	/**
	 * Logs a message under the WARNING LoggerLevel
	 * @param message The message to log
	 */
	public void warning(String message);
	
	/**
	 * Logs an exception under the WARNING LoggerLevel
	 * @param e The exception to log
	 */
	public void warning(Exception e);
	
	/**
	 * Logs a message under the ERROR LoggerLevel
	 * @param message The message to log
	 */
	public void error(String message);
	
	/**
	 * Logs an exception under the ERROR LoggerLevel
	 * @param e The exception to log
	 */
	public void error(Exception e);
	
	/**
	 * Logs a message under a given LoggerLevel
	 * @param message The message to log
	 * @param level The level to log under
	 */
	public void log(String message, LoggerLevel level);
	
	/**
	 * Logs a message under a given LoggerLevel
	 * @param e The exception to log
	 * @param level The level to log under
	 */
	public void log(Exception e, LoggerLevel level);

}

File: src/main/kotlin/uk/co/lukestevens/logging/Logger.kt
package uk.co.lukestevens.logging

/**
 * An interface to define a simple logger
 *
 * @author luke.stevens
 */
interface Logger {
	/**
	 * Logs a message under the INFO LoggerLevel
	 * @param message The message to log
	 */
	fun info(message: String)

	/**
	 * Logs an exception under the INFO LoggerLevel
	 * @param e The exception to log
	 */
	fun info(e: Exception)

	/**
	 * Logs a message under the DEBUG LoggerLevel
	 * @param message The message to log
	 */
	fun debug(message: String)

	/**
	 * Logs an exception under the DEBUG LoggerLevel
	 * @param e The exception to log
	 */
	fun debug(e: Exception)

	/**
	 * Logs a message under the WARNING LoggerLevel
	 * @param message The message to log
	 */
	fun warning(message: String)

	/**
	 * Logs an exception under the WARNING LoggerLevel
	 * @param e The exception to log
	 */
	fun warning(e: Exception)

	/**
	 * Logs a message under the ERROR LoggerLevel
	 * @param message The message to log
	 */
	fun error(message: String)

	/**
	 * Logs an exception under the ERROR LoggerLevel
	 * @param e The exception to log
	 */
	fun error(e: Exception)

	/**
	 * Logs a message under a given LoggerLevel
	 * @param message The message to log
	 * @param level The level to log under
	 */
	fun log(message: String, level: LoggerLevel)

	/**
	 * Logs a message under a given LoggerLevel
	 * @param e The exception to log
	 * @param level The level to log under
	 */
	fun log(e: Exception, level: LoggerLevel)
}